### ISSUE TITLE
x/exp/cmd/apidiff: incorrectly reports breaking change on numeric const using scientific e-notation

### ISSUE THREAD

**Issue by noahdietz** [golang/go#44796]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.16 darwin/amd64
</pre>

Also happens on `go version go1.16 linux/amd64`.

### Does this issue reproduce with the latest release?

Yes.

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GOARCH="amd64"
GOOS="darwin"
</pre></details>

Also happens on `linux/amd64`.

### What did you do?

```shell
go install golang.org/x/exp/cmd/apidiff@latest
git clone https://github.com/googleapis/google-cloud-go gocloud
cd gocloud/pubsub
apidiff -w pkg.master cloud.google.com/go/pubsub
apidiff -incompatible pkg.master cloud.google.com/go/pubsub 
```

### What did you expect to see?
Nothing, no breaking changes.

### What did you see instead?
```shell
- MaxPublishRequestBytes: value changed from 0.000582077 to 10000000
```

The LoC in question is [here](https://github.com/googleapis/google-cloud-go/blob/04d592f739d3f03a1c467de708222d0e502e961e/pubsub/topic.go#L49).

In short, it reads:

```go
const (
  MaxPublishRequestBytes = 1e7
)
```

We are trying to (re)enable `apidiff` in `cloud.google.com/go` CI, but can't enable it until this is fixed. We can always workaround this specific error though, I guess.

**Comment by ianlancetaylor** [comment]
CC @jba

**Comment by noahdietz** [comment]
I think it might be on the export side of things - because the "new" side of the diff outputs the proper value, while the "old" side  (I think loaded from the exported data) is reported as the mangled value. It could be that the constant value is incorrectly _written_ at export, or incorrectly _read_ at compat check. Since `apidiff` just invokes `gcexportdata` APIs, maybe there is an issue in there?

**Comment by noahdietz** [comment]
When I use a local install of `golang.org/x/exp/cmd/apidiff` with  no changes, there is no diff - it works properly. But when I install again via `go install golang.org/x/exp/cmd/apidiff@latest`, rewrite the base export data, and compare, it fails with the error above.

**Comment by noahdietz** [comment]
Updating the `golang.org/x/tools` dep to `v0.1.0` still _does not_ work. Going to check individual commits to `golang.org/x/tools` since the `v0.1.0` release that also changed the `gcexportdata` pkg to see if I can find the commit that fixes it. `v0.1.0` was cut on 1/19/21, so there are only a handful of commits to `gcexportdata` to check...

**Comment by noahdietz** [comment]
I checked out https://github.com/golang/tools/commit/a1db63cc08e0021aa688e1531b00ec88fd715898 locally (still wired with `replace`) and then `apidiff` worked properly. I have no idea how that would fix things.

Worth noting that some other deps in `exp` were updated as a result:
```
	golang.org/x/mod v0.4.1
	golang.org/x/sys v0.0.0-20210124154548-22da62e12c0c
	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
```

**Comment by jba** [comment]
It was fixed by commit b79f76fe3e9733fa292a2a558160b3658f45f187, CL b79f76fe3e9733fa292a2a558160b3658f45f187
https://go-review.googlesource.com/c/tools/+/292350. Anything past that should fix it.

**Comment by noahdietz** [comment]
Ok, good to know the exact fix, but that commit hasn't been released. Is it possible to cut a release of the root module in `tools`? Then we can update the version used in `exp` to capture the fix and `go install golang.org/x/exp/cmd/apidiff@latest` will work.

**Comment by noahdietz** [comment]
@ianthehat Hi would it be possible to cut a release of the `x/tools` root module? The commit jba mentioned that has a fix for `apidiff` that would unblock its use in cloud.google.com/go CI. If there is some regular release schedule, we can wait on that too! Thanks.
